engine/src/main/java/com/sandboni/core/engine/Application.java

```diff
@@ -56,6 +56,7 @@ Arguments buildArguments() {
                 .outputFormat(getValue(SystemProperties.OUTPUT_FORMAT))
                 .alwaysRunAnnotation(getValue(SystemProperties.ALWAYS_RUN_ANNOTATION))
                 .enablePreview(Boolean.parseBoolean(getValue(SystemProperties.ENABLE_PREVIEW)))
+                .runSelectiveModeOnReflectionCall(Boolean.parseBoolean(getValue(SystemProperties.SELECTIVE_MODE_REFLECTION)))
```

Comment:
why is it needed?

Reply:
So we can turn it off if/when we need to (or our users ask us)

Reply:
ok

Reply:
added, although when we turn this on, it will affect jar scanning, so I'm not sure if it's the best solution. please have a look.